server/channels/app/platform/session.go

```diff
@@ -16,6 +16,13 @@ import (
 func (ps *PlatformService) ReturnSessionToPool(session *model.Session) {
 	if session != nil {
 		session.Id = ""
+		// Once the session is retrieved from the pool, all existing prop fields are cleared.
+		// To avoid a race between clearing the props and accessing it, clear the props maps before returning it to the pool.
+		for k := range session.Props {
+			delete(session.Props, k) // clear is only available in go 1.21
+		}
+		// Also clear the team members slice to avoid a similar race condition.
+		session.TeamMembers = session.TeamMembers[:0]
```

Comment:
Sorry Ben, this PR is taking so long. But we can't simply reset the slice length without setting the elements to their default values, which in this case is `nil`, for them to be garbage collected.

https://go.dev/wiki/SliceTricks

> NOTE If the type of the element is a pointer or a struct with pointer fields, which need to be garbage collected, the above implementations of Cut and Delete have a potential memory leak problem: some elements with values are still referenced by slice a’s underlying array, just not “visible” in the slice. Because the “deleted” value is referenced in the underlying array, the deleted value is still “reachable” during GC, even though the value cannot be referenced by your code. If the underlying array is long-lived, this represents a leak.

Reply:
Yeah, I did read that before (https://stackoverflow.com/a/16972044/23659723), but I couldn't find a source that explains what is going on. Thanks for linking the wiki article.

I'm leaning toward removing the code that clears `session.TeamMembers`. I feel like we are trying to solve an issue that doesn't exist. The msgbuffer code clears `session.TeamMembers`. A similar can be said for `session.Props`. We are not fundamentally solving a race issue here, and we are just moving the race to a different place (as we saw with the failed tests on `master`)

Reply:
> I'm leaning toward removing the code that clears session.TeamMembers.

No objections to that. :+1: 

> A similar can be said for session.Props. We are not fundamentally solving a race issue here

Correct, but this is also about correctness and performance. Wiping off the map before returning to the pool is not in the hot path. And this should make unmarshalling the session faster because there are no map entries to clear off.

I would say let's remove the clearing of session.TeamMembers altogether because this is going out of scope and unrelated to the original problem. But would lean towards keeping the clearing of session.Props.